src/tracer.js

```diff
@@ -40,28 +42,30 @@ export default class Tracer {
     _tags: any;
     _injectors: any;
     _extractors: any;
+    _metrics: any;
 
     constructor(serviceName: string,
             reporter: Reporter = new NoopReporter(),
             sampler: Sampler = new ConstSampler(false),
-            logger: any,
-            tags: any = {}) {
-        this._tags = tags;
+            options: any = {}) {
+        this._tags = options.tags || {};
         this._tags[constants.JAEGER_CLIENT_VERSION_TAG_KEY] = `Node-${pjson.version}`;
         this._tags[constants.TRACER_HOSTNAME_TAG_KEY] = Utils.myIp();
 
+        this._metrics = options.metrics || new Metrics(new NoopMetricFactory());
+
         this._serviceName = serviceName;
         this._reporter = reporter;
         this._sampler = sampler;
-        this._logger = logger || new NullLogger();
+        this._logger = options.logger || new NullLogger();
         this._injectors = {};
         this._extractors = {};
 
-        let textCodec = new TextMapCodec(false);
+        let textCodec = new TextMapCodec(false, { metrics: this._metrics });
```

Comment:
I don't think that makes sense since the first parameter is required.  Generally only optional arguments get put into a json object.